src/DurableTask.Core/TaskOrchestrationDispatcher.cs

```diff
@@ -697,6 +697,11 @@ bool ReconcileMessagesWithState(TaskOrchestrationWorkItem workItem)
                     continue;
                 }
 
+                if (Activity.Current != null && message.Event is TimerFiredEvent)
+                {
+                    using Activity activity = TraceHelper.CreateActivityForTimer(workItem.OrchestrationRuntimeState.OrchestrationInstance, message.Event.Timestamp, ((TimerFiredEvent)message.Event).FireAt);
```

Comment:
nit: Now that I'm looking at this, it feels like a very unintuitive use of the C# `using` feature. I think it would be easier for readers of the to understand what we're doing if we disposed the activity explicitly, instead of implicitly. That also reduces the chance that someone accidentally introduces a bug by moving curly braces around. I also suggest a comment explaining what we're doing here since it's a bit counterintuitive.
```suggestion
                    // We immediately publish the activity span for this timer by creating the activity and immediately calling Dispose() on it.
                    TraceHelper.CreateActivityForTimer(workItem.OrchestrationRuntimeState.OrchestrationInstance, message.Event.Timestamp, ((TimerFiredEvent)message.Event).FireAt).Dispose();
```